### PR TITLE
[KERNEL32] FindFirstFileExW: Don't forget to release RTL_RELATIVE_NAME_U, and ignore lpSearchFilter correctly

### PR DESCRIPTION
**Pull request by RatinCN**
## Purpose

A little fixes for FindFirstFileExW behavior

JIRA issue: None

## Proposed changes

- Release `RTL_RELATIVE_NAME_U` before function return
- Ignore `lpSearchFilter` completely as well as Windows, MSDoc said `lpSearchFilter` must be `NULL`, but `FindFirstFileExW` will not check it

### CODE REVIEW

**Comment by HBelusca** [comment]
> Ignore lpSearchFilter completely as well as Windows, MSDoc said lpSearchFilter must be NULL, but FindFirstFileExW will not check it

How do you know that? Is there a test to prove this?

**Comment by RatinCN** [comment]
> 
Yes. Pass `L"XXX"` to `lpSearchFilter` and call it:
image

On my Win2K3:
image

On my Win11:
image

**Comment by RatinCN** [comment]
Yes, pass `L"XXX"` to `lpSearchFilter`:
```C
WIN32_FIND_DATA ffd;
HANDLE h = FindFirstFileExW(L".", FindExInfoStandard, &ffd, FindExSearchNameMatch, L"XXX", 0);
if (h != INVALID_HANDLE_VALUE)
{
    MessageBoxW(NULL, L"FindFirstFileExW Success!", L"Test", 0);
} else
{
    DWORD dw = GetLastError();
    MessageBoxW(NULL, L"FindFirstFileExW Fail!", L"Test", 0);
}
```
On Win2K3:
image

On Win11:
image

Did you mean this should be a test case and add to apitests? This is not a functional problem at all @HBelusca

File: dll/win32/kernel32/client/file/find.c [hunk 1]
```diff
@@ -692,13 +692,6 @@ FindFirstFileExW(IN LPCWSTR lpFileName,
         DECLSPEC_ALIGN(4) BYTE DirectoryInfo[FIND_DATA_SIZE];
         DIR_INFORMATION DirInfo = { .DirInfo = &DirectoryInfo };
 
-        /* The search filter is always unused */
-        if (lpSearchFilter)
```
Comment (SergeGautherie) [inline dll/win32/kernel32/client/file/find.c:696]: Add `UNREFERENCED_PARAMETER(lpSearchFilter);` at the start of the function.

**Review by HBelusca — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]
Fixes 6 test failures in the tests I have added in 017004a0ac9.

Before:
```
FindFiles: 367 tests executed (0 marked as todo, 16 failures), 0 skipped.
```

After:
```
FindFiles: 367 tests executed (0 marked as todo, 10 failures), 0 skipped.
```

**Pull request merged**